src/index.ts

```diff
@@ -0,0 +1,607 @@
+    try {
+      soosLogger.info("Starting SOOS DAST Analysis");
+      soosLogger.info(`Creating scan for project ${this.args.projectName}...`);
+      const result = await soosApiClient.createScan({
+        clientId: this.args.clientId,
+        projectName: this.args.projectName,
+        commitHash: this.args.commitHash,
+        branch: this.args.branchName,
+        buildVersion: this.args.buildVersion,
+        buildUri: this.args.buildUri,
+        branchUri: this.args.branchUri,
+        integrationType: this.args.integrationType,
+        operatingEnvironment: this.args.operatingEnvironment,
+        integrationName: this.args.integrationName,
+        appVersion: this.args.appVersion,
+        scriptVersion: null,
+        contributingDeveloperAudit: undefined,
+        scanType: ScanType.DAST,
+        toolName: CONSTANTS.DAST.TOOL,
+        toolVersion: CONSTANTS.DAST.TOOL_VERSION,
+      });
+      projectHash = result.projectHash;
+      branchHash = result.branchHash;
+      analysisId = result.analysisId;
+
+      soosLogger.info(`Checking if url '${this.args.targetURL}' is available...`);
+      if (!(await isUrlAvailable(this.args.targetURL))) {
+        soosLogger.error(`The URL ${this.args.targetURL} is not available.`);
+        exit(1);
+      }
+
+      execSync("mkdir -p ~/.ZAP/reports /root/.ZAP/reports");
+
+      if (this.args.reportRequestHeaders) {
+        execSync("cp -R /zap/reports/traditional-json-headers ~/.ZAP/reports/traditional-json");
+        execSync("cp -R /zap/reports/traditional-json-headers /root/.ZAP/reports/traditional-json");
+      } else {
+        execSync("cp -R /zap/reports/traditional-json ~/.ZAP/reports/traditional-json");
+        execSync("cp -R /zap/reports/traditional-json /root/.ZAP/reports/traditional-json");
+      }
+
+      const zapCommandGenerator = new ZAPCommandGenerator(this.args);
+      const command = zapCommandGenerator.runCommandGeneration(this.args.scanMode);
+      soosLogger.info(`Running command: ${command}`);
+      await SOOSDASTAnalysis.runZap(command);
+      const runSuccess = fs.existsSync(CONSTANTS.FILES.REPORT_SCAN_RESULT_FILE);
+      soosLogger.info(`Scan finished with success: ${runSuccess}`);
+
+      const discoveredUrls =
+        fs.existsSync(CONSTANTS.FILES.SPIDERED_URLS_FILE_PATH) &&
+        fs.statSync(CONSTANTS.FILES.SPIDERED_URLS_FILE_PATH).isFile()
+          ? fs
+              .readFileSync(CONSTANTS.FILES.SPIDERED_URLS_FILE_PATH, "utf-8")
+              .split("\n")
+              .filter((url) => url.trim() !== "")
+          : [];
+
+      const data = JSON.parse(fs.readFileSync(CONSTANTS.FILES.REPORT_SCAN_RESULT_FILE, "utf-8"));
+      data["discoveredUrls"] = discoveredUrls;
+      fs.writeFileSync(CONSTANTS.FILES.REPORT_SCAN_RESULT_FILE, JSON.stringify(data, null, 4));
+      const formData = new FormData();
+
+      formData.append("resultVersion", data["@version"]);
+      formData.append(
+        "file",
+        convertStringToB64(
```

Comment:
```suggestion
        convertStringToBase64(
```
(added comment on api client)